src/hotspot/cpu/aarch64/macroAssembler_aarch64.cpp

```diff
@@ -4263,8 +4263,7 @@ void MacroAssembler::load_nklass(Register dst, Register src) {
 
   // Check if we can take the (common) fast path, if obj is unlocked.
   ldr(dst, Address(src, oopDesc::mark_offset_in_bytes()));
-  tst(dst, markWord::monitor_value);
-  br(Assembler::EQ, fast);
+  tbz(dst, exact_log2(markWord::monitor_value), fast);
```

Comment:
This is a bit fragile because monitor_value must be a power of 2 for it to work.

Reply:
Yes, and it is (and hopefully, always will be - or else that code would need to be changed anyway). exact_log2() has an assert that checks exactly that.

Reply:
Missed the latter. Lgtm.

Reply:
one could even make monitor value and friends `const_expr` and use STATIC_ASSERT here. That way, you get a compile error if that assumption fails without paying at runtime.